lib/Doctrine/ORM/Query/AST/TypedExpression.php

```diff
@@ -0,0 +1,16 @@
+<?php
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Query\AST;
+
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * Provides an API for resolving the type of a Node
+ *
+ * @internal
```

Comment:
As mentioned in another comment we have extensions that could benefit from having DBAL handling the type conversion, so I think this doesn't necessarily need to be marked as `@internal`.

Reply:
This is fixed